polkadot/zombienet-sdk-tests/src/coretime-revenue/main.rs

```diff
@@ -0,0 +1,574 @@
+			let event = event.unwrap();
+			log::debug!("Got event: {} :: {}", event.pallet_name(), event.variant_name());
+			{
+				events.write().await.push((block.number().into(), event.clone()));
+			}
+			if event.pallet_name() == "Broker" {
+				match event.variant_name() {
+					"Purchased" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::Purchased>().unwrap().unwrap()
+					),
+					"SaleInitialized" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::SaleInitialized>().unwrap().unwrap()
+					),
+					"HistoryInitialized" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::HistoryInitialized>().unwrap().unwrap()
+					),
+					"CoreAssigned" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::CoreAssigned>().unwrap().unwrap()
+					),
+					"Pooled" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::Pooled>().unwrap().unwrap()
+					),
+					"ClaimsReady" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::ClaimsReady>().unwrap().unwrap()
+					),
+					"RevenueClaimBegun" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::RevenueClaimBegun>().unwrap().unwrap()
+					),
+					"RevenueClaimItem" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::RevenueClaimItem>().unwrap().unwrap()
+					),
+					"RevenueClaimPaid" => log::trace!(
+						"{:#?}",
+						event.as_event::<broker_events::RevenueClaimPaid>().unwrap().unwrap()
+					),
+					_ => (),
+				}
+			}
+		}
+	}
+}
+
+async fn wait_for_para_event<C: subxt::Config + Clone, E: StaticEvent, P: Fn(&E) -> bool + Copy>(
+	events: ParaEvents<C>,
+	pallet: &'static str,
+	variant: &'static str,
+	predicate: P,
+) -> E {
+	loop {
+		let mut events = events.write().await;
+		if let Some(entry) = events.iter().find(|&e| {
+			e.1.pallet_name() == pallet &&
+				e.1.variant_name() == variant &&
+				predicate(&e.1.as_event::<E>().unwrap().unwrap())
+		}) {
+			let entry = entry.clone();
+			events.retain(|e| e.0 > entry.0);
+			return entry.1.as_event::<E>().unwrap().unwrap();
+		}
+		drop(events);
+		tokio::time::sleep(std::time::Duration::from_secs(6)).await;
+	}
+}
+
+async fn ti_watcher<C: subxt::Config + Clone>(api: OnlineClient<C>, prefix: &'static str)
+where
+	<C::Header as subxt::config::Header>::Number: Display,
+{
+	let mut blocks_sub = api.blocks().subscribe_finalized().await.unwrap();
+
+	let mut issuance = 0i128;
+
+	log::debug!("Starting parachain watcher");
+	while let Some(block) = blocks_sub.next().await {
+		let block = block.unwrap();
+
+		let ti = api
+			.storage()
+			.at(block.reference())
+			.fetch(&rococo_api::storage().balances().total_issuance())
+			.await
+			.unwrap()
+			.unwrap() as i128;
+
+		let diff = ti - issuance;
+		if diff != 0 {
+			log::info!("{} #{} issuance {} ({:+})", prefix, block.number(), ti, diff);
+		}
+		issuance = ti;
+	}
+}
+
+// TODO: @s0me0ne-unkn0wn can we remove this fn
```

Comment:
I wanted to add some balance assertions later, so it's just a matter of preparation. But first, I need to sort out the problem with submitting the revenue claim transaction, which always fails without an obvious reason :/